src/aind_metadata_service/sharepoint/client.py

```diff
@@ -52,71 +47,190 @@ def __init__(
-        subject_procedures: None/list
+        subject_procedures: Optional[list]
+          An optional list of subject_procedures.
 
         Returns
         -------
-        int
-          Status code
-        Optional[BaseModel]
-          A Procedures model if valid, None otherwise.
+        Optional[Procedures]
+          A Procedures model if subject_procedures, else None.
 
         """
         if subject_procedures:
             procedures = Procedures.construct(subject_id=subject_id)
             procedures.subject_procedures = subject_procedures
-            response = Responses.model_response(procedures)
+            return procedures
+        else:
+            return None
+
+    @staticmethod
+    def _merge_procedures(
+        left_procedures: List[Procedures], right_procedures: List[Procedures]
+    ) -> List[Procedures]:
+        """
+        Merges two lists of Procedures. Given the way the lists are
+        constructed, the length of the lists will either be 0 or 1.
+        Parameters
+        ----------
+        left_procedures : List[Procedures]
+        right_procedures : List[Procedures]
+
+        Returns
+        -------
+        List[Procedures]
+          Single model with the subject_procedures merged from both lists.
+
+        """
+        if len(left_procedures) == 0:
+            return right_procedures
+        elif len(right_procedures) == 0:
+            return left_procedures
+        else:
+            subject_id = left_procedures[0].subject_id
+            new_subject_procedures = (
+                left_procedures[0].subject_procedures
+                + right_procedures[0].subject_procedures
+            )
+            return [
+                Procedures.construct(
+                    subject_id=subject_id,
+                    subject_procedures=new_subject_procedures,
+                )
+            ]
+
+    def _merge_two_responses(
+        self,
+        left_model_response: ModelResponse[Procedures],
+        right_model_response: ModelResponse[Procedures],
+    ) -> ModelResponse[Procedures]:
+        """
+        Merges two ModelResponses of Procedures type.
+        Parameters
+        ----------
+        left_model_response : ModelResponse[Procedures]
+        right_model_response : ModelResponse[Procedures]
+
+        Returns
+        -------
+        ModelResponse[Procedures]
+          A merged response.
+
+        """
+        if (
+            left_model_response.status_code.value >= 500
+            and right_model_response.status_code.value >= 500
+        ):
+            return ModelResponse.internal_server_error_response()
+        elif (
+            left_model_response.status_code == StatusCodes.DB_RESPONDED
+            and right_model_response.status_code.value >= 500
+        ):
+            return ModelResponse(
+                aind_models=left_model_response.aind_models,
+                status_code=StatusCodes.MULTIPLE_RESPONSES,
```

Comment:
We used StatusCodes.MULTI_STATUS when two responses with different status were merged, and used StatusCodes.MULTIPLE_RESPONSES when multiple subject items of the same subject id were returned from the database  .